### ISSUE TITLE
the extension in ios crashes the app on open not work.

### ISSUE THREAD

**Issue by rojinaghimire65** [defold/extension-facebook#64]
Just after including the extension in iOS crashes Defold.
iOS version: 18
The only log i can find is `'fbaemkit.framework/fbaemkit' (no such file)`

**Comment by britzl** [comment]
It could actually be that we need to add FBAEM to the Podspec:

https://github.com/defold/extension-facebook/blob/master/facebook/manifests/ios/Podfile

```
pod 'FBAEMKit', '17.0.0'
```

**Comment by britzl** [comment]
@rojinaghimire65 could you please try this one as a dependency in game.project: 

https://github.com/defold/extension-facebook/archive/refs/heads/Issue-64-missing-fbaemkit.zip

**Comment by rojinaghimire65** [comment]
@britzl 
Does not work. Still crashes.

**Comment by britzl** [comment]
> Still crashes.

What does the device log show?

**Comment by rojinaghimire65** [comment]
"basebandVersion" : "7.00.00",
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGABRT"},
  "termination" : {"code":1,"flags":518,"namespace":"DYLD","indicator":"Library missing","details":["(terminated at launch; ignore backtrace)"],"reasons":["Library not loaded: @rpath\/FBAEMKit.framework\/FBAEMKit","Referenced from: <F1CEACCD-C366-33A2-BE6F-A93577589CE6> \/Volumes\/VOLUME\/*\/Carrom.app\/Carrom","Reason: tried: '\/usr\/lib\/swift\/FBAEMKit.framework\/FBAEMKit' (no such file, not in dyld cache), '\/private\/preboot\/Cryptexes\/OS\/usr\/lib\/swift\/FBAEMKit.framework\/FBAEMKit' (no such file), '\/private\/var\/containers\/Bundle\/Application\/539DC39D-1E18-4B18-BF17-4FE43908E3C3\/Carrom.app\/Frameworks\/FBAEMKit.framework\/FBAEMKit' (mach-o file (\/private\/var\/containers\/Bundle\/Application\/539DC39D-1E18-4B18-BF17-4FE43908E3C3\/Carrom.app\/Frameworks\/FBAEMKit.framework\/FBAEMKit), but incompatible platform (have 'iOS-sim', need 'iOS')), '\/usr\/lib\/swift\/FBAEMKit.framework\/FBAEMKit' (no such file, not in dyld cache), '\/private\/preboot\/Cryptexes\/OS\/usr\/lib\/swift\/FBAEMKit.framework\/FBAEMKit' (no such file), '\/private\/var\/containers\/Bundle\/Application\/539DC39D-1E18-4B18-BF17-4FE43908E3C3\/Carrom.app\/Frameworks\/FBAEMKit.framewor"]},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":339604,"threadState":{"x":[{"value":6},{"value":1},{"value":6090566176},{"value":153},{"value":6090565152},{"value":0},{"value":0},{"value":635248},{"value":32},{"value":6090565021},{"value":10},{"value":0},{"value":50},{"value":0},{"value":10809720866},{"value":6090570842},{"value":521},{"value":7515963348,"symbolLocation":420,"symbol":"__simple_bprintf"},{"value":0},{"value":0},{"value":6090565152},{"value":153},{"value":6090566176},{"value":1},{"value":6},{"value":6090569144},{"value":6090568992},{"value":6090568952},{"value":0}],"flavor":"ARM_THREAD_STATE64","lr":{"value":7516398812},"cpsr":{"value":2147483648},"fp":{"value":6090565104},"sp":{"value":6090565040},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":7516371080,"matchesCrashFrame":1},"far":{"value":0}},"frames":[{"imageOffset":444552,"symbol":"__abort_with_payload","symbolLocation":8,"imageIndex":1},{"imageOffset":472284,"symbol":"abort_with_payload_wrapper_internal","symbolLocation":104,"imageIndex":1},{"imageOffset":472336,"symbol":"abort_with_payload","symbolLocation":16,"imageIndex":1},{"imageOffset":42332,"symbol":"dyld4::halt(char const*, dyld4::StructuredError const*)","symbolLocation":300,"imageIndex":1},{"imageOffset":90080,"symbol":"dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*)","symbolLocation":4124,"imageIndex":1},{"imageOffset":240120,"symbol":"dyld4::start(dyld4::KernelArgs*, void*, void*)::$_0::operator()() const","symbolLocation":544,"imageIndex":1},{"imageOffset":212144,"symbol":"start","symbolLocation":2188,"imageIndex":1}]}],

**Comment by britzl** [comment]
I tried bundling the project for iOS using the `Issue-64-missing-fbaemkit` branch and inspect the generated .app and the build/arm64-ios folders and I see the FBAEMKit.framework:

<img width="294" alt="Screenshot 2024-12-20 at 06 55 14" src="https://github.com/user-attachments/assets/f00b6b33-ab44-4a8d-8f15-275df510e98d" />

<img width="362" alt="Screenshot 2024-12-20 at 06 54 45" src="https://github.com/user-attachments/assets/399b2e25-ab2f-4f27-ab03-5d1e41613698" />

Can you please check your bundle too?

**Comment by rojinaghimire65** [comment]
It looks same and still crashes. @britzl

**Comment by britzl** [comment]
> It looks same and still crashes. @britzl

Does it crash immediately upon launch or are you calling any particular FB function?

**Comment by kindablissy** [comment]
Yes it does. Just adding it as a dependency crashes the game immediately on launch.

**Comment by rojinaghimire65** [comment]
@britzl

**Comment by britzl** [comment]
> @britzl

I'm away on Christmas vacation and the iPhone I have access to here can't be updated to iOS 18.

Could you please run the Console.app and capture all device logs from before launching the app until it crashes? You can send the log to bjorn@defold.se if you are not comfortable sharing the entire log here. I will take a look and see if there's anything that stands out. That is the best I can do for now.

**Comment by britzl** [comment]
> can't be updated to iOS 18.

By the way, you specifically mention iOS 18. Does the same app work on iOS 17 and below?

**Comment by rojinaghimire65** [comment]
mailed you from rojinaghimire6538@gmail.com @britzl
